platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/TaskList.java

```diff
@@ -210,4 +271,106 @@ AtomicInteger getAttemptsLeft(Operation op) {
   Map<Operation, Map<String, Predicate<ProfileMigrationReport>>> getTasks() {
     return groups;
   }
+
+  @SuppressWarnings("squid:S1452" /* Using wildcards to simplify testing */)
+  @VisibleForTesting
```

Comment:
❓ Instead of having all those `@VisibleForTesting` methods, couldn't we test using side-effects? For instance, create mock tasks and compound tasks and make sure they are or not called? Not sure we can do that for all of them, but if we can eliminate as many as we can that would decouple the tests from the implementation a little bit more.

Reply:
I toyed with that early on.
Since this class is really used at two different times by a user: one to add tasks and one to execute them. It felt easier to test the class in each of these phases separately. 
I do agree it ties us a little bit more to the implementation though and I am also not found of exposing its core like that.